### PR TITLE
fix(storage): respect STORAGE_EMULATOR_HOST in signedURL

### PR DESCRIPTION
**Pull request by ChrisBr**
Respect STORAGE_EMULATOR_HOST in signedURL.

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -308,7 +308,11 @@ type bucketBoundHostname struct {
...
 func (s pathStyle) host(bucket string) string {
-	return "storage.googleapis.com"
+	if os.Getenv("STORAGE_EMULATOR_HOST") {
+		return os.Getenv("STORAGE_EMULATOR_HOST")
+	} else {
+		return "storage.googleapis.com"
+	}
 }
 
 func (s virtualHostedStyle) host(bucket string) string {
```
Comment (tritone) [inline storage/storage.go:318]: The path for virtual hosted should be updated as well probably?

File: storage/storage.go [hunk 2]
```diff
@@ -308,7 +308,11 @@ type bucketBoundHostname struct {
 }
 
 func (s pathStyle) host(bucket string) string {
-	return "storage.googleapis.com"
+	if os.Getenv("STORAGE_EMULATOR_HOST") {
+		return os.Getenv("STORAGE_EMULATOR_HOST")
```
Comment (tritone) [inline storage/storage.go:312]: We accept STORAGE_EMULATOR_HOST both with and without a scheme, so we should strip the scheme if it is present. See https://github.com/googleapis/google-cloud-go/blob/main/storage/storage.go#L154

**Review by tritone — COMMENTED** [review]
Thanks for the PR. A couple thoughts; let me know if you are able to fix or if you'd like some help. It would also be good to add a test case for this to TestSignedURLV4.

**Comment by BrennaEpp** [comment]
Hi @ChrisBr, thanks for your contribution! I made a PR to your branch to incorporate the changes suggested by @tritone. Can you look that over and sign the CLA as well? 

To sign the CLA, please visit https://cla.developers.google.com/. Once you've signed, click here https://cla.developers.google.com/github/rescan?pr=googleapis%2Fgoogle-cloud-go%2Fpull%2F5411&key=%01~%D4%13%80JC%15G%B8Y%92%FC%87%D9%82%D4%CE%97%DC%40&expiry=2022-03-15 to update the check

**Comment by ChrisBr** [comment]
Sorry for the delay, will check tomorrow with the CLA 👍

**Comment by ChrisBr** [comment]
The CLA is still under review with my employer Shopify but I will hopefully have an answer in the next couple of days. Sorry for the delay.

**Comment by ChrisBr** [comment]
@BrennaEpp signed the CLA 👍 Thanks for your patience.

**Comment by BrennaEpp** [comment]
These changes were merged in #5673. Thank you for your contribution!

**Pull request closed without merging**